code/ATMOSPHERICS/hvac/spaceheater.dm

```diff
@@ -38,25 +68,38 @@
 		overlays  += "[base_state]-open"
 	return
 
+/obj/machinery/space_heater/campfire/update_icon()
+	overlays.len = 0
+	if(on)
+		var/fireintensity = min(Floor((cell.charge-1)/(cell.maxcharge/4))+1,4)
+		icon_state = "[base_state][fireintensity]"
+		light_range_on = fireintensity+1
+		light_power_on = 0.4*fireintensity
```

Comment:
@dylanstrategie nerd it will work, `light_range` is the one that only works with integers, not `light_power`.